tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/util/K8sUtil.java

```diff
@@ -336,11 +310,44 @@ public static void executeInPod(
                 releaseName);
     }
 
+    /**
+     * Returns the default labels defined for a Platform Operator pod.
+     *
+     * @param releaseName the release name of the Platform Operator chart
+     * @return {@link Map}  Map containing some of the fixed labels expected for a Platform Operator pod.
+     */
+    public static @NotNull Map<String, String> getHiveMQPlatformOperatorLabels(final @NotNull String releaseName) {
+        return Map.of("app.kubernetes.io/instance", releaseName, "app.kubernetes.io/name", "hivemq-platform-operator");
+    }
+
+    /**
+     * Returns the expected init container with as per the given initContainerName from the given
+     * {@link StatefulSetSpec}
+     * instance.
+     * <p>
+     * If the init container with the given name is not found, a {@link java.util.NoSuchElementException} will be
+     * thrown.
+     *
+     * @param statefulSetSpec   the {@link StatefulSetSpec} to retrieve the HiveMQ container from
+     * @param initContainerName the name of the container to look for
+     * @return the container
+     */
+    public static @NotNull Container getInitContainer(
+            final @NotNull StatefulSetSpec statefulSetSpec, final @NotNull String initContainerName) {
+        return statefulSetSpec.getTemplate()
+                .getSpec()
+                .getInitContainers()
+                .stream()
+                .filter(container -> container.getName().equals(initContainerName))
+                .findFirst()
+                .orElseThrow();
+    }
+
     /**
      * Returns a Predicate condition based on the state of a Kubernetes resource.
      */
     @SuppressWarnings("unchecked")
-    public static @NotNull Predicate<GenericKubernetesResource> getHiveMQPlatformStatus(final @NotNull String state) {
+    public static @NotNull Predicate<GenericKubernetesResource> getKubernetesResourceStatus(final @NotNull String expectedState) {
```

Comment:
I find this method name a bit misleading, since it will only work for K8s resources that have a `state` field in their status. Without weird coincidences this will only apply to the legacy and platform custom resources.

Maybe `getCustomResourceStateCondition()` is a bit more precise?